extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -0,0 +1,90 @@
+import { ExtensionContext } from 'vscode';
+import { Credentials } from '../authentication';
+import { Logger } from '../logging';
+import { downloadArtifactFromLink } from './gh-actions-api-client';
+import { DownloadLink } from './download-link';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { AnalysisSummary } from './shared/remote-query-result';
+import * as sarif from 'sarif';
+import { AnalysisResults, QueryResult } from './shared/analysis-result';
+
+export class AnalysesResultsManager {
+  // Store for the results of various analyses for a single remote query.
+  private readonly analysesResults: { [key: string]: QueryResult[] };
+
+  constructor(
+    private readonly ctx: ExtensionContext,
+    private readonly logger: Logger,
+  ) {
+    this.analysesResults = {};
+  }
+
+  public async downloadAnalysisResults(
+    analysisSummary: AnalysisSummary,
+  ): Promise<void> {
+    const credentials = await Credentials.initialize(this.ctx);
+
+    void this.logger.log(`Downloading and processing results for ${analysisSummary.nwo}`);
+
+    const queryResults = await this.downloadSingleAnalysisResults(
+      analysisSummary.downloadLink,
+      credentials);
+
+    this.analysesResults[analysisSummary.nwo] = queryResults;
+  }
+
+  public async downloadAllResults(
+    analysisSummaries: AnalysisSummary[],
+  ): Promise<void> {
+    const credentials = await Credentials.initialize(this.ctx);
+
+    void this.logger.log('Downloading and processing all results');
+
+    for (const analysis of analysisSummaries) {
+      const queryResults = await this.downloadSingleAnalysisResults(analysis.downloadLink, credentials);
+      this.analysesResults[analysis.nwo] = queryResults;
+    }
+  }
+
+  public getFlattenedAnalysesResults(): AnalysisResults[] {
```

Comment:
nit: Since the only way to interact with `AnalysisResults` is as an array, perhaps you can rename this to:

```suggestion
  public getAnalysesResults(): AnalysisResults[] {
```

Reply:
Renamed. I think ideally we should be returning a deep copy of the array to make sure that the manager is the only entity manipulating the data but I haven't seen other places in the code doing that so I've not introduced anything here. Perhaps something to discuss though.